web/src/main/java/org/springframework/security/web/authentication/rememberme/TokenBasedRememberMeServices.java

```diff
@@ -83,22 +96,34 @@
  */
 public class TokenBasedRememberMeServices extends AbstractRememberMeServices {
 
+	private RememberMeHashingAlgorithm hashingAlgorithm;
+
 	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService) {
+		this(key, userDetailsService, RememberMeHashingAlgorithm.UNSET);
+	}
+
+	/**
+	 * @since 5.5
+	 */
+	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService,
+			RememberMeHashingAlgorithm hashingAlgorithm) {
 		super(key, userDetailsService);
+		this.hashingAlgorithm = hashingAlgorithm;
 	}
```

Comment:
We should prevent the algorithm from being null in the constructor

```suggestion
	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService) {
		super(key, userDetailsService);
	}

	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService, RememberMeHashingAlgorithm hashingAlgorithm) {
		this(key, userDetailsService);
		Assert.notNull(hashingAlgorithm, "hashingAlgorithm cannot be null");
		this.hashingAlgorithm = hashingAlgorithm;
	}
```